### PR TITLE
Prep 0.94.0 release

### LINKED ISSUE #2610: Release 0.94.0
### Component(s)

_No response_

### Describe the issue you're reporting

Operator 0.94.0 should be released. 

@TylerHelmuth you are the next on the list https://github.com/open-telemetry/opentelemetry-operator/blob/main/RELEASE.md#release-schedule. 

However we should wait until we have mo work done for v1alpha2 CRD

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Fixes #2610

### CODE REVIEW

**Review by frzifus — APPROVED** [review]

**Review by yuriolisa — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -40,10 +40,10 @@ The operator should be released within a week after the [OpenTelemetry collector
...
 | Version  | Release manager |
 |----------|-----------------|
-| v0.94.0  | @TylerHelmuth   |
 | v0.95.0  | @swiatekm-sumo  |
 | v0.96.0  | @frzifus        |
 | v0.97.0  | @jaronoff97     |
 | v0.98.0  | @yuriolisa      |
 | v0.99.0  | @pavolloffay    |
 | v0.100.0 | @VineethReddy02 |
+| v0.101.0 | @TylerHelmuth   |
```
Comment (pavolloffay) [inline RELEASE.md:49]: 🔨

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,37 @@ Changes by Version
...
+
+### Components
+
+* [OpenTelemetry Collector - v0.94.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.94.0)
+* [OpenTelemetry Contrib - v0.94.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.94.0)
+* [Java auto-instrumentation - 1.32.1](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.32.1)
+* [.NET auto-instrumentation - 1.2.0](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/1.2.0)
+* [Node.JS - 0.46.0](https://github.com/open-telemetry/opentelemetry-js-contrib/releases/tag/auto-instrumentations-node-0.46.0)
+* [Python - 0.43b0](https://github.com/open-telemetry/opentelemetry-python-contrib/releases/tag/0.43b0)
+* [Go - v0.10.0-alpha](https://github.com/open-telemetry/opentelemetry-go-instrumentation/releases/tag/v0.10.0-alpha)
```
Comment (pavolloffay) [inline CHANGELOG.md:31]: 0.10.1-alpha is used

**Review by pavolloffay — COMMENTED** [review]
Please update https://github.com/open-telemetry/opentelemetry-operator/blob/main/hack/add-openshift-annotations.sh#L3. It should be 4.12-4.15

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,37 @@ Changes by Version
...
+* [ApacheHTTPD - 1.0.4](https://github.com/open-telemetry/opentelemetry-cpp-contrib/releases/tag/webserver%2Fv1.0.4)
+* [Nginx - 1.0.4](https://github.com/open-telemetry/opentelemetry-cpp-contrib/releases/tag/webserver%2Fv1.0.4)]
```
Comment (pavolloffay) [inline CHANGELOG.md:33]: the ending `]` should be removed

**Review by swiatekm-sumo — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
> Please update https://github.com/open-telemetry/opentelemetry-operator/blob/main/hack/add-openshift-annotations.sh#L3. It should be 4.12-4.15

@pavolloffay so I can update the release process, how did you know the version that needed to be

File: versions.txt [hunk 4]
```diff
@@ -2,20 +2,20 @@
 # by default with the OpenTelemetry Operator. This would usually be the latest
 # stable OpenTelemetry version. When you update this file, make sure to update the
 # the docs as well.
-opentelemetry-collector=0.93.0
+opentelemetry-collector=0.94.0
```
Comment (jaronoff97) [inline versions.txt:5]: i think the collector is at 0.94.1 https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.94.1

Comment (TylerHelmuth) [reply versions.txt:5]: The libraries are, but not the released images: https://github.com/open-telemetry/opentelemetry-collector-releases/releases

**Pull request merged**